### ISSUE TITLE
[Web] Transformed widget is affected by another widget's borderRadius or circular shape   

### ISSUE THREAD

**Issue by egorbelibov** [flutter/flutter#48631]
### Issue
There's some **Black Magic** 🧙‍♂️affecting a transformed container. It's caused by adding a `borderRadius` or `BoxShape.circle` to a **separate** widget in a `Stack`. 

### Flutter Version
```console
[✓] Flutter (Channel master, v1.13.9-pre.57, on Mac OS X 10.15.2 19C57, locale en-PT) 
```

## Code to Reproduce
1. Create new **project**.
2. Replace with the attached code.
3. De-comment `IssueSource` commented code. 

<details>
	<summary>code sample</summary>

```dart
import 'package:flutter/material.dart';

void main() => runApp(MyApp());

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      home: TestPage(),
    );
  }
}

class TestPage extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return Scaffold(
      body: Column(
        mainAxisAlignment: MainAxisAlignment.center,
        crossAxisAlignment: CrossAxisAlignment.center,
        children: <Widget>[
          TroublesomeContainer(),
        ],
      ),
    );
  }
}

// Massive issue caused by this guy's container's [Radius] or [BoxShape.circle] 🤬
class IssueSource extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return Positioned(
      top: 0,
      left: 0,
      child: Container(
        width: 35,
        height: 35,
        decoration: BoxDecoration(
          color: Color(0xFF000000),
          // ------------------------------------------------------------------
          // MASSIVE ISSUE: When de-commented the transformed widget is affected
          // ------------------------------------------------------------------
          // shape: BoxShape.circle,
          //
          // or de-comment this one
          // borderRadius: BorderRadius.all(Radius.circular(20))
        ),
      ),
    );
  }
}

class TroublesomeContainer extends StatelessWidget {
  Widget build(BuildContext context) {
    return _renderButton();
  }

  Widget _renderButton() {
    return Expanded(
      child: Stack(
        fit: StackFit.expand,
        children: [
          _transformedContainer(
            child: AffectedContainer(),
          ),
          IssueSource(),
        ],
      ),
    );
  }

  Widget _transformedContainer({@required Widget child}) {
    return Align(
      alignment: Alignment.center,
      child: Container(
        width: 500,
        height: 500,
        child: Transform(
          alignment: Alignment.center,
          transform: Matrix4.skew(0.50, 0.50)
            ..add(Matrix4.identity()
              ..setEntry(3, 2, 0.001)
              ..rotateX(0.90)
              ..rotateZ(0.90)),
          child: child,
        ),
      ),
    );
  }
}

class AffectedContainer extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return Center(
      child: Container(
        decoration: BoxDecoration(
          // ISSUE: Border Radius causes widget to disappear from screen.
          // borderRadius: BorderRadius.all(
          // Radius.circular(5.0),
          // ),
          color: Color(0xFF000000),
        ),
        child: Center(
          child: Row(
            children: [
              Center(
                child: Container(
                  decoration: BoxDecoration(
                    // ISSUE: Border Radius causes widget to disappear from screen.
                    // borderRadius: BorderRadius.all(
                    //   Radius.circular(2.0),
                    // ),
                    color: Color(0xFFFFFFFF),
                  ),
                  width: 10,
                  height: 10,
                ),
              ),
            ],
          ),
        ),
      ),
    );
  }
}
```

</details>

### Screenshots
- **Without** borderRadius / circular shape decoration:

![image](https://github.com/egorbelibov/github_assets/blob/master/flutter/screenshots/issues/48631/transformed_widget_without_borderRadius.png?raw=true)

- **With** borderRadius / circular shape decoration:

![image](https://github.com/egorbelibov/github_assets/blob/master/flutter/screenshots/issues/48631/transformed_widget_with_borderRadius.png?raw=true)

**Comment by egorbelibov** [comment]
@ferhatb I think this might be **related**, **but not the same** issue addressed by the following PR : #15327

**Comment by TahaTesser** [comment]
Hi @egorbelibov 
This seems to be fixed, I cannot reproduce it on the latest `master`

| Without BorderRadius | with BorderRadius |
| --------------- | --------------- |
<img width="320" alt="Screenshot 2020-11-11 at 6 49 05 PM" src="https://user-images.githubusercontent.com/48603081/98816587-e8664e80-244e-11eb-829d-5398fa51e07c.png"> | <img width="320" alt="Screenshot 2020-11-11 at 6 49 18 PM" src="https://user-images.githubusercontent.com/48603081/98816615-f1efb680-244e-11eb-96f2-05d312a4ecab.png">


<details> 
<summary>code sample</summary> 

```dart
import 'package:flutter/material.dart';

void main() => runApp(MyApp());

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      home: TestPage(),
    );
  }
}

class TestPage extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return Scaffold(
      body: Column(
        mainAxisAlignment: MainAxisAlignment.center,
        crossAxisAlignment: CrossAxisAlignment.center,
        children: <Widget>[
          TroublesomeContainer(),
        ],
      ),
    );
  }
}

// Massive issue caused by this guy's container's [Radius] or [BoxShape.circle] 🤬
class IssueSource extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return Positioned(
      top: 0,
      left: 0,
      child: Container(
        width: 35,
        height: 35,
        decoration: BoxDecoration(
          color: Color(0xFF000000),
          // ------------------------------------------------------------------
          // MASSIVE ISSUE: When de-commented the transformed widget is affected
          // ------------------------------------------------------------------
          shape: BoxShape.circle,
        ),
        //
        // or de-comment this one
        // borderRadius: BorderRadius.all(Radius.circular(20))),
      ),
    );
  }
}

class TroublesomeContainer extends StatelessWidget {
  Widget build(BuildContext context) {
    return _renderButton();
  }

  Widget _renderButton() {
    return Expanded(
      child: Stack(
        fit: StackFit.expand,
        children: [
          _transformedContainer(
            child: AffectedContainer(),
          ),
          IssueSource(),
        ],
      ),
    );
  }

  Widget _transformedContainer({@required Widget child}) {
    return Align(
      alignment: Alignment.center,
      child: Container(
        width: 500,
        height: 500,
        child: Transform(
          alignment: Alignment.center,
          transform: Matrix4.skew(0.50, 0.50)
            ..add(Matrix4.identity()
              ..setEntry(3, 2, 0.001)
              ..rotateX(0.90)
              ..rotateZ(0.90)),
          child: child,
        ),
      ),
    );
  }
}

class AffectedContainer extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return Center(
      child: Container(
        decoration: BoxDecoration(
          // ISSUE: Border Radius causes widget to disappear from screen.
          // borderRadius: BorderRadius.all(
          // Radius.circular(5.0),
          // ),
          color: Color(0xFF000000),
        ),
        child: Center(
          child: Row(
            children: [
              Center(
                child: Container(
                  decoration: BoxDecoration(
                    // ISSUE: Border Radius causes widget to disappear from screen.
                    // borderRadius: BorderRadius.all(
                    //   Radius.circular(2.0),
                    // ),
                    color: Color(0xFFFFFFFF),
                  ),
                  width: 10,
                  height: 10,
                ),
              ),
            ],
          ),
        ),
      ),
    );
  }
}

```

</details>


<details> 
<summary>flutter doctor -v</summary> 

```bash
[✓] Flutter (Channel master, 1.24.0-8.0.pre.194, on Mac OS X 10.15.7 19H15
    darwin-x64, locale en-GB)
    • Flutter version 1.24.0-8.0.pre.194 at /Users/tahatesser/Code/flutter_master
    • Framework revision 018467cdb1 (4 hours ago), 2020-11-11 02:04:03 -0500
    • Engine revision 81f219c59c
    • Dart version 2.12.0 (build 2.12.0-31.0.dev)

[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.2)
    • Android SDK at /Users/tahatesser/Code/sdk
    • Platform android-30, build-tools 30.0.2
    • ANDROID_HOME = /Users/tahatesser/Code/sdk
    • Java binary at: /Applications/Android
      Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build
      1.8.0_242-release-1644-b3-6915495)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 12.2)
    • Xcode at /Volumes/SanDisk/Xcode-beta.app/Contents/Developer
    • Xcode 12.2, Build version 12B5044c
    • CocoaPods version 1.10.0

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 4.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build
      1.8.0_242-release-1644-b3-6915495)

[✓] VS Code (version 1.51.0)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.16.0

[✓] Connected device (4 available)
    • Taha’s iPad (mobile) • 00008020-000255113EE8402E • ios            • iOS 14.2
    • macOS (desktop)      • macos                     • darwin-x64     • Mac OS X
      10.15.7 19H15 darwin-x64
    • Web Server (web)     • web-server                • web-javascript • Flutter
      Tools
    • Chrome (web)         • chrome                    • web-javascript • Google
      Chrome 86.0.4240.193

• No issues found!
```

</details>

Closing, as fixed. If you disagree, please write in the comments and I will reopen it.
Thank you
